src/client/theme-default/index.ts

```diff
@@ -22,7 +23,10 @@ export { default as VPTeamMembers } from './components/VPTeamMembers.vue'
 
 const theme: Theme = {
   Layout,
-  NotFound
+  NotFound,
+  enhanceApp: ({ app }) => {
```

Comment:
As there was [a discussion here](https://github.com/vuejs/vitepress/pull/1239/files#r958301764), we still have a problem doing this since now when extending the default theme, we can't simply do this

```ts
import DefaultTheme from 'vitepress/theme'

export default {
  ...DefaultTheme,
  enhanceApp({ app }) {
    // register global components
    app.component('MyGlobalComponent', /* ... */)
  }
}
```

The above `enhanceApp` will override `VPBadge` registration... 🤔 Hmmm...

I really like the above way of extending the default theme because it's super simple.

@brc-dd @meteorlxy What do you think if we make `VPBadge` non default, but let users include it when they need it.

```html
<!-- Basic usage would be manual import like this -->
<script>
import { VPBadge } from 'vitepress/theme'
</script>

### Title <VPBadge ...>
```

And document if they want to use this often, extend default theme and register it in `enhanceApp` 👀 

```ts
// Let users do this if user wants to use it often.
export default {
  ...DefaultTheme,
  enhanceApp: ({ router, app }) => {
    app.component('Badge', VPBadge)
  }
}
```

Reply:
I'm with brc-dd what we could not always avoid enhanceApp in default theme. We can also consider providing a util function for that, something like:

```ts
export default useDefaultTheme({
  enhanceApp: () => {
  // user's own enhanceApp
  }
})
```

Reply:
We've discussed this on Discord too. We should provide better extending API. though we should do that on another PR so for now, with this PR, I'll document the extend section to do:

```ts
import DefaultTheme from 'vitepress/theme'

export default {
  ...DefaultTheme,
  enhanceApp(ctx) {
    DefaultTheme.enhanceApp(ctx)
    ctx.app.component('MyGlobalComponent', /* ... */)
  }
}
```